integration_test.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"math/rand"
+	"os"
+	"os/exec"
+	"testing"
+	"text/template"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/api/iterator"
+)
+
+// integration test validates the end-to-end fluentbit and bigquery pipeline
+func TestPipeline(t *testing.T) {
+	logFilePath := "./logfile.log"
+	configFilePath := "./fluent-bit.conf"
+	ctx := context.Background()
+
+	// Set up BigQuery client
+	client, err := bigquery.NewClient(ctx, "bigquerytestdefault")
+	if err != nil {
+		t.Fatalf("Failed to create BigQuery client: %v", err)
+	}
+	defer client.Close()
+
+	// Create a random table name
+	hash := randString()
+	tableID := "test_" + hash
+
+	// Create BigQuery table in an existing dataset and project
+	datasetID := "siddag_summer2024"
+	tableSchema := bigquery.Schema{
+		{Name: "Message", Type: bigquery.StringFieldType},
+	}
+	dataset := client.Dataset(datasetID)
+	table := dataset.Table(tableID)
+	if err := table.Create(ctx, &bigquery.TableMetadata{Schema: tableSchema}); err != nil {
+		t.Fatalf("Failed to create BigQuery table: %v", err)
+	}
+
+	//Create config file with random table name
+	if err := createConfigFile(tableID); err != nil {
+		t.Fatalf("failed to create config file: %v", err)
+	}
+
+	// Create log file
+	file, err := os.Create(logFilePath)
+	if err != nil {
+		t.Fatalf("Failed to create log file: %v", err)
+	}
+	defer file.Close()
+
+	// Start Fluent Bit with the config file
+	FBcmd := exec.Command("fluent-bit", "-c", "fluent-bit.conf")
+	if err := FBcmd.Start(); err != nil {
+		t.Fatalf("Failed to start Fluent Bit: %v", err)
+	}
+
+	//Wait for fluent-bit connection to generate data; add delays before ending fluent-bit process
+	time.Sleep(2 * time.Second)
+	if err := generateData(10); err != nil {
+		t.Fatalf("Failed to generate data: %v", err)
+	}
+	time.Sleep(2 * time.Second)
+
+	// Stop Fluent Bit
+	if err := FBcmd.Process.Kill(); err != nil {
```

Comment:
Just curious, does this shutdown method also result in the same error message we saw earlier from the managed writer?

Reply:
I'm not completely sure because the errors logged to stdout aren't shown when we start and stop fluent-bit like this (I think because there is no standard output where the errors should be logged specified). I can maybe look into this in the future.

Reply:
No problem.